SAMS-57 update content

Lisa-Stubert opened the pull request:
I have added all the new content. This also includes:

- more consistent gendering and thus adaptation of titles and also paths to module 1 (see second last commit)
- a small list with two bullet points in the intro text on the title page. Please have a look if css is ok.
- The xxx's have been replaced by the word 'ServiceAgent:innen'. I styled this word in italics because it makes sense to me that way in the reading flow. Please check if this is ok style-wise. @dnsos @vogelino 
- the links did not work, I made a small fix (see first commit)

Three small "problems" have appeared:
- In two cases two sources are given for one link. I have now simply selected one of the sources and entered it as a link. 
- The modules 4,5,6 and 8 have no methods. I have therefore removed the method from the FrontMatter. This creates much whitespace between the example and the link list.
- In the module overview page the short texts of the single modules are partly a bit too long. For module 8, the text even flows out of the box.

Lisa-Stubert commented:
@dnsos @vogelino Theoretically rdy for review, but the test failed. Can this be due to the fact that I have removed the "Methods" for some modules (because they have no methods)?

dnsos commented:
@Lisa-Stubert the tests were failing because not all tests were in sync with the new content (see my most recent commit). It's a bit annoying having to update the texts in the tests manually. Automating that is not really priority, but could be done if we have some time left.

I'll take a look at the content changes with a fresh mind tomorrow.

dnsos commented on src/_includes/partials/links_list.liquid:
```diff
@@ -8,7 +8,7 @@
       <a
         class="group-odd:bg-blue-50 group-odd:bg-opacity-50 p-4 pr-16 sm:p-6 md:p-8 sm:text-lg md:text-xl grid grid-cols-1 md:grid-cols-[1fr,2fr] lg:grid-cols-[2fr,3fr,1fr] gap-2 sm:gap-8 transition group-hover:text-magenta-500 relative"
-        href="{{ link.url }}"
+        href="{{ link.link }}"
```
Good catch. This was not a smart naming strategy from my side. Feel free to improve this, e.g. by actually changing the fields in `further_reading` to `url` or the variable `link` to `resource`

dnsos reviewed (changes requested):
Thanks for including all this content @Lisa-Stubert!

I have only a few small comments, take a look at them and see what you want to improve in this PR. Concerning the whitespace created by missing methods: Let's only render the `methods` partial when there are methods:

```liquid
{% if methods %}
{% include 'partials/methods.liquid' %}
{% endif %}
```

This should remove the vertical padding that feels a bit too much.

I wouldn't worry about the texts that overflow the module card. We should fix this by CSS, so this is not really an issue of this PR. I would suggest adding a separate ticket for this.

> In two cases two sources are given for one link. I have now simply selected one of the sources and entered it as a link.

👍

vogelino reviewed (commented):
Thank you for taking care of that! I have no further comments than @dnsos. Good job!

Lisa-Stubert commented:
Thanks for the review and tipps! I think I implemented anything. I added a Tickets for the overlapping text, but I can't reproduce the problem to make a screenshot.. Have u already fixed it?

dnsos reviewed (approved):
Great!

About the overflowing texts: I have reduced the line height of the magenta-colored numbers which might have also "solved" the text problem. Let's review this separately and merge this!
